### DISCUSSION TITLE
How Can I Help?

### DISCUSSION THREAD

**Discussion by wjallison** [DFHack/dfhack#2494] [Q&A]
I'm an engineer with moderate programming experience who has some time off coming up, and I would like to learn about memory hacking. How can I get started?

**Comment by myk002** [comment]
There's a lot going on with DFHack to get involved with!

Are you specifically looking to get into the reverse engineering? If so, that's not really a short term commitment kind of thing. However, if that's where your interest lies, I'd suggest following the #reverse-engineering channel on our discord server: https://dfhack.org/discord

There are, however, many other things that you can do. They are also more likely to give you an opportunity to provide real value to the project. Our high level roadmap is here: https://github.com/orgs/DFHack/projects/23

As you can see, the current priority items are in the Lua space and center on reliability and user experience. If you've never seen the Lua language before, I can vouch that it's a delight to use and isn't too hard to learn.

But anyway, come to discord and let's talk!

**Comment by ab9rf** [comment]
if you're specifically interested in reverse engineering technologies and are fairly competent with java, we have a reverse engineering script in java that needs some love. some of the changes are fairly straightforward, others are more complicated. if you're interested, find me on our discord.
